BlockchainSdk/Blockchains/TON/TONTransactionBuilder.swift

```diff
@@ -61,36 +69,91 @@ final class TONTransactionBuilder {
     ///   - destination: Destination address transaction
+    ///   - walletAddress: Address of jetton wallet, required for jetton transaction
     /// - Returns: TheOpenNetworkSigningInput for sign transaction with external signer
-    private func input(amount: Amount, destination: String, params: TONTransactionParams?) throws -> TheOpenNetworkSigningInput {
-        let transfer = try self.transfer(amount: amount, destination: destination, params: params)
-        
-        // Sign input with dummy key of Curve25519 private key
-        let input = TheOpenNetworkSigningInput.with {
-            $0.transfer = transfer
-            $0.privateKey = inputPrivateKey.rawRepresentation
+    private func input(
+        amount: Amount,
+        destination: String,
+        walletAddress: String? = nil,
+        params: TONTransactionParams?
+    ) throws -> TheOpenNetworkSigningInput {
+        switch amount.type {
+        case .coin, .reserve:
+            let transfer = try transfer(amountValue: amount.value, destination: destination, params: params)
+            
+            // Sign input with dummy key of Curve25519 private key
+            return TheOpenNetworkSigningInput.with {
+                $0.transfer = transfer
+                $0.privateKey = inputPrivateKey.rawRepresentation
+            }
+        case .token(let token):
+            guard let walletAddress else {
+                fatalError("Wallet address must be set for jetton trasaction")
+            }
+            let transfer = try jettonTransfer(
+                amount: amount,
+                destination: destination,
+                walletAddress: walletAddress,
+                token: token,
+                params: params
+            )
+
+            // Sign input with dummy key of Curve25519 private key
+            return TheOpenNetworkSigningInput.with {
+                $0.jettonTransfer = transfer
+                $0.privateKey = inputPrivateKey.rawRepresentation
+            }
         }
-        
-        return input
     }
     
     /// Create transfer message transaction to blockchain
     /// - Parameters:
     ///   - amount: Amount transaction
     ///   - destination: Destination address transaction
     /// - Returns: TheOpenNetworkTransfer message for Input transaction of TON blockchain
-    private func transfer(amount: Amount, destination: String, params: TONTransactionParams?) throws -> TheOpenNetworkTransfer {
+    private func transfer(
+        amountValue: Decimal,
+        destination: String,
+        params: TONTransactionParams?
+    ) throws -> TheOpenNetworkTransfer {
         TheOpenNetworkTransfer.with {
             $0.walletVersion = TheOpenNetworkWalletVersion.walletV4R2
             $0.dest = destination
-            $0.amount = ((amount.value * wallet.blockchain.decimalValue) as NSDecimalNumber).uint64Value
+            $0.amount = ((amountValue * wallet.blockchain.decimalValue) as NSDecimalNumber).uint64Value
```

Comment:
Можно так, есть extension под это 
```suggestion
            $0.amount = (amountValue * wallet.blockchain.decimalValue).uint64Value
```

Reply:
округление надо добавить тут? Иначе при дробном значении можем получить 0

Reply:
ну вообще не должно быть дробного, при умножении на `decimalValue` должно целочисленное получается )

Reply:
поправил